var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -456,6 +456,7 @@ def cmake_args(self):
 
         python = spec['python']
         cmake_args = [
+            define("CMAKE_BUILD_RPATH", os.path.join(self.build_directory, 'lib')),
```

Comment:
I think this needs to be the install rpath *plus* the build lib. Otherwise you'll potentially get missing shared libs from upstream dependencies. If the cmake install rpath isn't accessible as a property we should make it accessible.

Reply:
Yeah, I had hoped CMAKE_BUILD_USE_INSTALL_RPATH would append paths :D it doesn't do that. Okay, I'll just merge those paths by hand then.

Reply:
@sethrj I doubt this is required though. Do you remember why you ever needed CMAKE_BUILD_USE_INSTALL_RPATH? I checked a simple cmake example where I'm setting CMAKE_BUILD_PATH and find_package(ZLIB) with zlib from Spack, and it adds rpaths in the elf files for both zlib and CMAKE_BUILD_RPATH. I kinda want to know why we're managing rpaths during the build ourselves... I still hope it's all redundant.

Reply:
Lol @sethrj, for the longest time I believed you contributed this line, but you only refactored it, sorry for the noise...